src/hotspot/share/utilities/elfFile.cpp

```diff
@@ -91,7 +91,7 @@ int FileReader::read_buffer(void* buf, size_t size) {
   return fread(buf, 1, size, _fd);
 }
 
-bool FileReader::set_position(long offset) {
+bool FileReader::set_position(int64_t offset) {
```

Comment:
You introduce a bug here.
`fseek` declaration:
```
int fseek ( FILE * stream, long int offset, int origin );
```

`fseek` will read only 32 bits of `offset` if `sizeof(long)==32`.